File: Data Structures & Algorithms/eating-bananas/submission-0.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int result_rate;
        int left = 1;
        int max_val;

        for (auto& p : piles){
            max_val = max(p, max_val);
        }      

        int right = max_val;
        int total_time;

        while( left <= right){
            int mid = ( left + right ) / 2;
            total_time = 0;
            for (auto& p: piles){
                total_time += ceil((double)p / mid);
            }
            if (total_time <= h){
                result_rate = mid;
                right = mid - 1;
            }
            else {
                left = mid + 1;
            }
        }
        return result_rate;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.res = 0
        
        def findNodes(maxVal, node):
            if not node:
                return
            if node.val >= maxVal:
                self.res +=1
            findNodes(max(maxVal, node.val), node.left)
            findNodes(max(maxVal, node.val), node.right)
        
        findNodes(root.val, root)

        return self.res
                
            
            
        
        
        